js/link_test.ts

```diff
@@ -63,7 +63,18 @@ testPerm({ read: true, write: true }, function linkSyncNotFound(): void {
   assertEquals(err.name, "NotFound");
 });
 
-test(function linkSyncPerm(): void {
+testPerm({ read: false, write: true }, function linkSyncReadPerm(): void {
```

Comment:
Just to have full coverage, could you add test case with `{ read: true, write: false }`? That applies to `rename` tests as well

Reply:
@bartlomieju It's at the bottom of this diff :) Unless you meant `{ read: false, write: false }`?

Reply:
My bad! Then LGTM :+1: